provider/ultradns_test.go

```diff
@@ -0,0 +1,658 @@
+		},
+	}
+	sbpoolRDataList := []udnssdk.SBRDataInfo{}
+	changes := &plan.Changes{}
+	changes.UpdateNew = []*endpoint.Endpoint{{DNSName: "kubernetes-ultradns-provider-test.com.", Targets: endpoint.Targets{"1.1.2.2", "192.168.0.24"}, RecordType: "A", RecordTTL: 100}}
+	changesList := &UltraDNSChanges{
+		Action: "UPDATE",
+		ResourceRecordSetUltraDNS: udnssdk.RRSet{
+			RRType:    "A",
+			OwnerName: "kubernetes-ultradns-provider-test.com.",
+			RData:     []string{"1.1.2.2", "192.168.0.24"},
+			TTL:       100,
+		},
+	}
+
+	for _, _ = range changesList.ResourceRecordSetUltraDNS.RData {
+
+		rrdataInfo := udnssdk.SBRDataInfo{
+			RunProbes: true,
+			Priority:  1,
+			State:     "NORMAL",
+			Threshold: 1,
+			Weight:    nil,
+		}
+		sbpoolRDataList = append(sbpoolRDataList, rrdataInfo)
+	}
+	sbPoolObject := udnssdk.SBPoolProfile{
+		Context:     udnssdk.SBPoolSchema,
+		Order:       "ROUND_ROBIN",
+		Description: "kubernetes-ultradns-provider-test.com.",
+		MaxActive:   2,
+		MaxServed:   2,
+		RDataInfo:   sbpoolRDataList,
+		RunProbes:   true,
+		ActOnProbes: true,
+	}
+
+	actualSBPoolObject, _ := provider.newSBPoolObjectCreation(context.Background(), changesList)
+	assert.Equal(t, sbPoolObject, actualSBPoolObject)
+
+}
+
+//Testcase to check fail scenario for multiple AAAA targets
+func TestUltraDNSProvider_MultipleTargetAAAA(t *testing.T) {
+	_, ok := os.LookupEnv("ULTRADNS_INTEGRATION")
+	if !ok {
+		log.Printf("Skipping test")
+
+	} else {
+
+		provider, _ := NewUltraDNSProvider(endpoint.NewDomainFilter([]string{"kubernetes-ultradns-provider-test.com"}), false)
+		changes := &plan.Changes{}
+		changes.Create = []*endpoint.Endpoint{
+			{DNSName: "ttl.kubernetes-ultradns-provider-test.com", Targets: endpoint.Targets{"2001:0db8:85a3:0000:0000:8a2e:0370:7334", "2001:0db8:85a3:0000:0000:8a2e:0370:7335"}, RecordType: "AAAA", RecordTTL: 100},
+		}
+		err := provider.ApplyChanges(context.Background(), changes)
+		if err == nil {
+			t.Errorf("We wanted it to fail since multiple AAAA targets are not allowed")
+		}
+		resp, _ := provider.client.Do("GET", "https://api.ultradns.com/zones/kubernetes-ultradns-provider-test.com./rrsets/AAAA/ttl.kubernetes-ultradns-provider-test.com.", nil, udnssdk.RRSetListDTO{})
+		assert.Equal(t, resp.Status, "404 Not Found")
+	}
+}
+
+// Test case to check multiple CNAME targets.
+func TestUltraDNSProvider_MultipleTargetCNAME(t *testing.T) {
+	_, ok := os.LookupEnv("ULTRADNS_INTEGRATION")
+	if !ok {
+		log.Printf("Skipping test")
+
+	} else {
+		provider, err := NewUltraDNSProvider(endpoint.NewDomainFilter([]string{"kubernetes-ultradns-provider-test.com"}), false)
+		changes := &plan.Changes{}
+
+		changes.Create = []*endpoint.Endpoint{
+			{DNSName: "ttl.kubernetes-ultradns-provider-test.com", Targets: endpoint.Targets{"nginx.loadbalancer.com.", "nginx1.loadbalancer.com."}, RecordType: "CNAME", RecordTTL: 100},
+		}
+		err = provider.ApplyChanges(context.Background(), changes)
+		if err == nil {
```

Comment:
Please use assert.NotNil instead of this if condition. Please take care of this everywhere in udnssdk project as well.